### DISCUSSION TITLE
Im making recording program, but the program record too fast so the output video has shorter playtime than actual recording time.

### DISCUSSION THREAD

**Discussion by parkjongyoung7** [community/community#136924] [Programming Help]
### Select Topic Area

Question

### Body

 i record for 10 sec, but output video has only about 2 or 3 sec long. could anybody give a solution for this? thx!
here is the code

import pygetwindow as gw
import mss
import threading
import numpy as np
import cv2
import time

recording = True
VIDEO_OUTPUT_FILENAME = "output_video.mp4"
FPS = 144  

def record_video():
    global recording

    with mss.mss() as sct:
        monitor = sct.monitors[1]  
        monitor_bbox = (monitor['left'], monitor['top'], monitor['width'], monitor['height'])

        print("Recording video...")

        fourcc = cv2.VideoWriter_fourcc(*"H264")
        out = cv2.VideoWriter(VIDEO_OUTPUT_FILENAME, fourcc, 144, 
                              (monitor_bbox[2] - monitor_bbox[0], monitor_bbox[3] - monitor_bbox[1]))

        while recording:
            img = np.array(sct.grab(monitor_bbox))
            frame = cv2.cvtColor(img, cv2.COLOR_BGRA2BGR)
            out.write(frame)
            time.sleep(1 / FPS)

    out.release()
    print(f"Video saved to {VIDEO_OUTPUT_FILENAME}")

**Comment by hellorei** [comment]
You are likely having issues with your frame rate, recording at high frame rates can easily drop frames due to performance reasons.

You can try to debug your function by:

1. try with a lower resolution video
2. progressively lower frame rates (120 > 90 > 60 > 30) and see if that is the issue
3. try another codec.

Also, make sure the drive you are recording to is fast enough for the video resolution and encoding, higher frame rates and higher resolutions create larger amounts of data to be written, which require faster media.

I hope this helps.

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @parkjongyoung7 ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.
